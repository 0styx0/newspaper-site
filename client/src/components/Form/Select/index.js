import React from 'react';
import Label from '../Label';

import './index.css';

/**
 * @prop label - string, contents of label element
 * @prop required - if it exists, the select gets `required attribute
 * @prop name - name of select
 * @prop className - class
 * @prop children - option elements
 */
export default function Select(props) {

    const select = React.cloneElement(<select/>, props.props);

    return (
        <Label
            value={props.label}
            required={!!props.props.required}
            children={
                select
            }
        />
    );
}

