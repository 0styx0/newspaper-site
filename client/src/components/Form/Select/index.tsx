import * as React from 'react';

import Label from '../Label';

import './index.css';

interface Props {
    label: string;
    props: any; // really should be html attributes
}

/**
 * @prop label - string, contents of label element
 * @prop props - object of attributes of select
 */
export default function Select(props: Props) {

    const select = React.cloneElement(<select/>, props.props);

    return (
        <Label
            value={props.label}
            required={!!props.props.required}
            children={
                select
            }
        />
    );
}
