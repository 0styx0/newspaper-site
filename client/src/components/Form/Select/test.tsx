import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Select from './';

describe('<Select>', () => {

    describe('snapshots', () => {

        it('renders correctly', () => {

            const tree = renderer.create(
                <Select
                  label="Label"
                  props={{
                      name: 'hello',
                      children: [<option key="1" value="hi">Hi</option>]
                  }}
                />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});