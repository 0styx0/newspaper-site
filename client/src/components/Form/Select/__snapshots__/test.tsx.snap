// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> snapshots renders correctly 1`] = `
<label>
  Label: 
  <select
    name="hello"
  >
    <option
      value="hi"
    >
      Hi
    </option>
  </select>
  
</label>
`;
